src/Messages/OWSIdentityManager.m

```diff
@@ -385,18 +407,25 @@ - (void)createIdentityChangeInfoMessageForRecipientId:(NSString *)recipientId
 {
     OWSAssert(recipientId != nil);
 
+    NSMutableArray<TSMessage *> *messages = [NSMutableArray new];
+
     TSContactThread *contactThread = [TSContactThread getOrCreateThreadWithContactId:recipientId];
     OWSAssert(contactThread != nil);
 
     TSErrorMessage *errorMessage =
         [TSErrorMessage nonblockingIdentityChangeInThread:contactThread recipientId:recipientId];
-    [errorMessage save];
-
+    [messages addObject:errorMessage];
     [[TextSecureKitEnv sharedEnv].notificationsManager notifyUserForErrorMessage:errorMessage inThread:contactThread];
 
     for (TSGroupThread *groupThread in [TSGroupThread groupThreadsWithRecipientId:recipientId]) {
-        [[TSErrorMessage nonblockingIdentityChangeInThread:groupThread recipientId:recipientId] save];
+        [messages addObject:[TSErrorMessage nonblockingIdentityChangeInThread:groupThread recipientId:recipientId]];
     }
+
+    [self.storageManager.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+        for (TSMessage *message in messages) {
+            [message saveWithTransaction:transaction];
```

Comment:
I save everything here (and below) in a single transaction to reduce the number of "yap database modified" notifications.